src/utility.ts

```diff
@@ -281,11 +285,25 @@ export class Utility {
             } else {
                 deviceType = "device";
             }
-            device.iconPath = context.asAbsolutePath(path.join("resources", `${deviceType}-${state}.svg`));
+            if (context) {
+                device.iconPath = context.asAbsolutePath(path.join("resources", `${deviceType}-${state}.svg`));
+            }
             return device;
         });
     }
 
+    public static async getNoneEdgeDeviceIdList(iotHubConnectionString: string): Promise<string[]> {
+        const registry: Registry = Registry.fromConnectionString(iotHubConnectionString);
+        const query = registry.createQuery("SELECT * FROM DEVICES where capabilities.iotEdge=false");
+        const noneEdgeDevices = ((await query.nextAsTwin(null)) as ResultWithIncomingMessage<Twin[]>).result;

```

Comment:
Line 296~298 is similar to Line 427~ 429, could we wrap them into a method so that it could be shared? 
E.g. getDeviceList(iotHubConnectionString: string, isEdge: boolean)

Reply:
fixed